dotnet-engine/Yggdrasil.Engine/NativeLoader.cs

```diff
@@ -0,0 +1,179 @@
+
+        if (!File.Exists(tempPath))
+        {
+            using (var stream = Assembly.GetExecutingAssembly().GetManifestResourceStream($"{assemblyName}.{libName}"))
+            {
+                if (stream == null)
+                    throw new FileNotFoundException($"Embedded resource {libName} not found.");
+
+                using (var fileStream = new FileStream(tempPath, FileMode.Create, FileAccess.Write))
+                {
+                    stream.CopyTo(fileStream);
+                }
+            }
+        }
+
+        return LoadBinary(tempPath);
+    }
+
+    private static string GetBinaryName()
+    {
+        string os, arch, libc = "";
+
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            os = "win";
+        else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            os = "linux";
+        else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            os = "osx";
+        else
+            throw new PlatformNotSupportedException("Unsupported OS");
+
+        if (RuntimeInformation.ProcessArchitecture == Architecture.X64)
+            arch = "x86_64";
+        else if (RuntimeInformation.ProcessArchitecture == Architecture.Arm64)
+            arch = "arm64";
+        else if (RuntimeInformation.ProcessArchitecture == Architecture.X86)
+            arch = IntPtr.Size == 4 ? "i686" : "x86_64";
+        else
+            throw new PlatformNotSupportedException("Unsupported CPU architecture");
+
+        if (os == "linux" && IsMusl())
+            libc = "-musl";
+
+        string filename = os == "win"
+            ? $"yggdrasilffi_{arch}.dll"
+            : $"libyggdrasilffi_{arch}{libc}.{(os == "osx" ? "dylib" : "so")}";
+
+        return filename;
+    }
+
+    private static bool IsMusl()
+    {
+        if (!RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            return false;
+
+        try
+        {
+            string output = File.ReadAllText("/proc/self/maps");
+            return output.Contains("musl");
+        }
+        catch
+        {
+            try
+            {
+                using (var process = new System.Diagnostics.Process())
+                {
+                    process.StartInfo = new System.Diagnostics.ProcessStartInfo
+                    {
+                        FileName = "ldd",
+                        Arguments = "--version",
+                        RedirectStandardOutput = true,
+                        UseShellExecute = false
+                    };
+                    process.Start();
+                    string lddOutput = process.StandardOutput.ReadToEnd();
+                    process.WaitForExit();
+
+                    return lddOutput.Contains("musl");
+                }
+            }
+            catch
+            {
+                return false;
+            }
+        }
+    }
+
+    internal static IntPtr LoadFunctionPointer(IntPtr libHandle, string functionName)
```

Comment:
This method and the ones below start by trying to tap into the binary in the sane .NET core method and if we fail then we're probably running .NET framework and we fallback to kernel32.dll or libdl